src/pages/private/user/index.js

```diff
@@ -1,14 +1,24 @@
 // outsource dependencies
-import React from 'react';
-import PropTypes from 'prop-types';
-import { useDispatch } from 'react-redux';
+import React, { useMemo } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
 import { Button, ListGroup, ListGroupItem } from 'reactstrap';
 
 // local dependencies
 import { TYPE } from '../reducer';
+import { selector } from '../../reducer';
 
-function User ({ user }) {
+function User () {
   const dispatch = useDispatch();
+  const state = useSelector(selector);
+
+  const user = useMemo(() => {
+    if (state.user instanceof Object) {
+      const coverImage = state.user.coverImage instanceof Object
+        ? state.user.coverImage : {};
+      return { ...state.user, coverImage };
+    }
+    return {};
+  }, [state.user]);
```

Comment:
How do you think, is it possible to create separated local constant to define "avatar" "userId" and "userName" ?